### PR TITLE
HIP-584: Provide implementation for create token logic

### LINKED ISSUE #6340: HIP-584: Copy CreateLogic
### Problem

We need to copy `CreateLogic` from `hedera-services`, so that we can implement `CreatePrecompile`.

### Solution

Copy CreateLogic from `hedera-services` into `com.hedera.services` package in `hedera-mirror-web3`.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by mustafauzunn**
**Description**:

We need default implementations for precompiles related to creation of a new token, so we should copy needed logic from hedera-services into hedera-mirror-web3.

**Related issue(s)**:

Fixes #6340 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java [hunk 1]
```diff
@@ -77,6 +77,8 @@ public class MirrorNodeEvmProperties implements EvmProperties {
     @Getter
     private int maxGasEstimateRetriesCount = 20;
 
+    private int maxCustomFeesAllowed = 10;
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/MirrorNodeEvmProperties.java:80]: Sort alphabetically and add this property in docs/configuration.md

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/CreateLogic.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+import com.hedera.services.store.models.Id;
+import com.hedera.services.store.models.Token;
+import com.hedera.services.store.models.TokenRelationship;
+import com.hedera.services.txns.validation.OptionValidator;
+import com.hedera.services.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.TokenCreateTransactionBody;
+import java.util.List;
+import org.hyperledger.besu.datatypes.Address;
+
+public class CreateLogic {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/CreateLogic.java:47]: We should make the Logic type a stateless bean and add javadoc, documenting what are the differences compared to `hedera-services` repo.

Comment (stoyanov-st) [reply hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/CreateLogic.java:56]: Done

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by steven-sheehy** [comment]
Does this fix https://github.com/hashgraph/hedera-mirror-node/issues/5649 ? I marked it as fixing, so please fix to the correct one if not.

**Comment by IvanKavaldzhiev** [comment]
There wasn't an issue created, so I created a suitable one.

**Review by steven-sheehy — APPROVED** [review]
LGTM just need to merge to fix conflicts

**Review by IvanKavaldzhiev — APPROVED** [review]

**Pull request merged**